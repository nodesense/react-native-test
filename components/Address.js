import React from "react";

import {View, Text, Button} from "react-native";

export default class Address extends React.Component {

    constructor() {
        super();

        this.state = {
            show: true
        }
    }

    toggle() {
        console.log("**TOGGLE**");

        this.setState({
            show: !this.state.show
        });
    }

    render() {


        return (
            <View>
             {
                 this.state.show && (
                    <View>
                        <Text> Street: {this.props.address.street} </Text>
                        <Text> City: {this.props.address.city} </Text>
                        <Text> Pincode: {this.props.address.pincode}  </Text>
                        <Button title="Contact"
                                onPress= { () => {}} >
                        </Button>
                    </View>
              )
             }

                <Button title="Toggle"
                        onPress= { () => this.toggle()} >
                </Button>

            </View>
        )
    }
}