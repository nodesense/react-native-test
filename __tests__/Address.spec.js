// for JSX
import React from "react";

import Address from "../components/Address";

import TestRenderer from "react-test-renderer";

describe("Address comp test suite", () => {

    it("should render default props", () => {
        let address = {street: '1st', city: 'Nagpur', pincode: 100000};

        const renderer = TestRenderer.create(<Address address={address} />);

        const jsonOutput = renderer.toJSON();
        console.log("JSON ", jsonOutput);

        expect(jsonOutput).toMatchSnapshot();


        const root = renderer.root;

        const addressInst = root.instance;

        expect(addressInst.state.show).toBe(true);


        addressInst.toggle();

        expect(addressInst.state.show).toBe(false);


        expect(renderer.toJSON()).toMatchSnapshot();

    });

})