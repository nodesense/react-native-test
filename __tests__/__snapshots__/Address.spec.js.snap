// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Address comp test suite should render default props 1`] = `
<View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
       Street: 
      1st
       
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
       City: 
      Nagpur
       
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
       Pincode: 
      100000
        
    </Text>
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={undefined}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Contact
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={undefined}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Toggle
      </Text>
    </View>
  </View>
</View>
`;

exports[`Address comp test suite should render default props 2`] = `
<View>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={undefined}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Toggle
      </Text>
    </View>
  </View>
</View>
`;
